src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2941,6 +2941,18 @@ private bool isExactType(CORINFO_CLASS_STRUCT_* cls)
             return _compilation.IsEffectivelySealed(type);
         }
 
+        private TypeCompareState isNullableType(CORINFO_CLASS_STRUCT_* cls)
+        {
+            TypeDesc type = HandleToObject(cls);
+
+            if (type.IsCanonicalDefinitionType(CanonicalFormKind.Any))
```

Comment:
```suggestion
            if (type.IsGenericParameter)
```
`CanonicalFormKind.Any` is too conservative. `CanonicalFormKind.Specific` cannot ever be Nullable. On the other hand, it would be nice if this logic was same as isEnum logic below. The isEnum logic below does not handle UniversalCanon (it is ok - we have number of places that do not handle universal generics, universal generics are not used currently).

Reply:
Updated to `IsGenericParameter`, thanks.

Going by https://github.com/dotnet/runtime/pull/99555#discussion_r1520479299, would this be hit for something like this?
```c#
class C<TFrom>
{
    [NoInlining]
    public TTo BitCast<TTo>(TFrom t) => Unsafe.BitCast<TFrom, TTo>(t);
}

_ = new C<int?>().BitCast<long>(42);
```

Reply:
> Updated to IsGenericParameter, thanks.

The IsGenericParameter in the isEnum case is bogus and will never hit. I'm going to delete it in #101040 once all the legs finish, but it's already clear it never hits. Jit should never be asking about things that report true to `IsGenericParameter` (not in this API and not anywhere else).

It will never hit here either.

Reply:
Changed it to an assert now